quinn-proto/src/endpoint.rs

```diff
@@ -168,6 +169,7 @@ pub struct ListenKeys {
     ///
     /// Initialize with random bytes.
     pub reset: [u8; 64],
+    pub reset_key: SigningKey,
```

Comment:
I'm not a huge fan of having both of these fields in the struct, especially with no mechanism to prevent a struct from being constructed where their values are unrelated. Perhaps make `reset` private, and we can bug upstream to impl `Clone` for `SigningKey` to make this better in the future?

Reply:
Maybe a better approach would be to get rid of `reset` and modify the tests to generate new `ListenKeys` on demand.